gpiozero/mixins.py

```diff
@@ -270,13 +280,119 @@ def _fire_events(self):
             if not old_state and new_state:
```

Comment:
Not quite, I was proposing:
```python
if old_state is None:
    # initial state
elif old_state != new_state:
    if new_state:
        # state change to active
    else:
        # state change to inactive
```
i.e. _exactly_ the same logic as before, just restructured slightly. Apologies for not explaining it very well.